curp/src/server.rs

```diff
@@ -0,0 +1,75 @@
+use std::{collections::HashMap, sync::Arc};
+
+use parking_lot::Mutex;
+
+use crate::{
+    cmd::{Command, CommandExecutor},
+    message::{Propose, ProposeError, ProposeResponse},
+    util::MutexMap,
+};
+
+/// The server that handles client request and server consensus protocol
+#[derive(Debug, Clone)]
+pub struct Server<C: Command, CE: CommandExecutor<C>> {
+    /// The speculative cmd pool
+    spec: Arc<Mutex<HashMap<C::K, Arc<C>>>>,
+    /// Command executor
+    cmd_exeutor: Arc<CE>,
+}
+
+impl<C: 'static + Command, CE: 'static + CommandExecutor<C>> Default for Server<C, CE> {
+    #[inline]
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[allow(missing_docs)]
```

Comment:
already has docs?

Reply:
It's macro expand code, we can't modify it in this crate.